pkg/apis/stable/v1alpha1/gameserver.go

```diff
@@ -330,16 +332,23 @@ func (gs *GameServer) GetDevAddress() (string, bool) {
 	return devAddress, hasDevAddress
 }
 
-// IsAllocated returns true if the server is currently allocated.
-func (gs *GameServer) IsAllocated() bool {
-	return gs.ObjectMeta.DeletionTimestamp == nil && gs.Status.State == GameServerStateAllocated
+// IsDeletable returns false if the server is currently allocated/reserved and should not be deleted
+func (gs *GameServer) IsDeletable() bool {
+	return !(gs.ObjectMeta.DeletionTimestamp.IsZero() &&
```

Comment:
Not sure I understand the condition here, how about this for readability:

```
switch {
  case !gs.ObjectMeta.DeletionStamp.IsZero(): return false // already deleted
  case gs.Status.State == GameServerAllocated: return false
  case gs.Status.State == GameServerReserved: return false
  default: return true
}
```

Reply:
This got a little more nuanced, to ensure it kept passing the tests for `computeReconciliationAction` - PTAL, let me know if this new format is more readable though.